ddtrace/contrib/rq/__init__.py

```diff
@@ -0,0 +1,245 @@
+    with pin.tracer.trace("rq.queue.fetch_job", service=trace_utils.int_service(pin, config.rq)) as span:
+        job_id = get_argument_value(args, kwargs, 0, "job_id")
+        span.set_tag("job.id", job_id)
+        return func(*args, **kwargs)
+
+
+@trace_utils.with_traced_module
+def traced_perform_job(rq, pin, func, instance, args, kwargs):
+    """Trace rq.Worker.perform_job"""
+    # `perform_job` is executed in a freshly forked, short-lived instance
+    job = get_argument_value(args, kwargs, 0, "job")
+
+    ctx = HTTPPropagator.extract(job.meta)
+    if ctx.trace_id:
+        pin.tracer.context_provider.activate(ctx)
+
+    try:
+        with pin.tracer.trace(
+            "rq.worker.perform_job",
+            service=trace_utils.ext_service(pin, config.rq_worker),
+            span_type=SpanTypes.WORKER,
+            resource=job.func_name,
+        ) as span:
+            span.set_tag("job.id", job.get_id())
+            return func(*args, **kwargs)
+    finally:
+        if job.get_status() == rq.job.JobStatus.FAILED:
+            span.error = 1
+            span.set_tag("error.stack", job.exc_info)
+        span.set_tag("status", job.get_status())
+        span.set_tag("origin", job.origin)
+
+        # Force flush to agent since the process `os.exit()`s
+        # immediately after this method returns
+        pin.tracer.writer.flush_queue()
+
+
+@trace_utils.with_traced_module
+def traced_job_perform(rq, pin, func, instance, args, kwargs):
+    """Trace rq.Job.perform(...)"""
+    job = instance
+
+    # Inherit the service name from whatever parent exists.
+    # eg. in a worker, a perform_job parent span will exist with the worker
+    #     service.
+    with pin.tracer.trace("rq.job.perform", resource=job.func_name) as span:
+        span.set_tag("job.id", job.get_id())
+        return func(*args, **kwargs)
+
+
+@trace_utils.with_traced_module
+def traced_job_fetch(rq, pin, func, instance, args, kwargs):
+    """Trace rq.Job.fetch(...)"""
+
+    job = None
+    try:
+        with pin.tracer.trace("rq.job.fetch", service=trace_utils.ext_service(pin, config.rq_worker)) as span:
+            job_id = get_argument_value(args, kwargs, 0, "id")
+            span.set_tag("job.id", job_id)
+            job = func(*args, **kwargs)
+            return job
+    finally:
+        if job:
+            job_status = job.get_status()
+            span.set_tag("job.status", job_status)
+
+
+@trace_utils.with_traced_module
+def traced_job_fetch_many(rq, pin, func, instance, args, kwargs):
+    """Trace rq.Job.fetch_many(...)"""
+    with pin.tracer.trace("rq.job.fetch_many", service=trace_utils.ext_service(pin, config.rq_worker)) as span:
+        job_ids = get_argument_value(args, kwargs, 0, "job_ids")
+        span.set_tag("job_ids", job_ids)
+        return func(*args, **kwargs)
+
+
+def patch():
+    # Avoid importing rq at the module level, eventually will be an import hook
+    import rq
```

Comment:
so with this approach we'd end up with an import error on calling `patch` rather than an attribute error from `patch` no existing?